### DISCUSSION TITLE
Evaluation with CropForegroundd

### DISCUSSION THREAD

**Discussion by mikiwang820** [Project-MONAI/MONAI#1529] [Q&A]
Hello,

I used "CropForegroundd" in the validation transform when I was training the model, so I still applied it when I did evaluation and saved my predictive labels.
However, it caused the outputs had the different sizes from the ground truth.
Therefore, I take it out, and the outputs' sizes are right, but the dice score drop from 0.6 to 0.4.

How can I deal with this problem.
Thank you for any help you can provide

**Comment by rijobro** [comment]
You should be able to apply the same transform to both your ground truth and your segmentation. That way they'll both be the same size. Something like this:

```python
CropForegroundd(keys=["image","label"], source_key="label")
```

Here, both `image` (your ground truth) and the `label` (your segmentation) will be cropped using a lambda (by default voxels >0) applied on the `source_key` image (`label`).

**Reply by rijobro** [reply]
Check [this tutorial](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d.ipynb) for an example of `CropForegroundd`.

**Reply by FrancescoLR** [reply]
Hello, I am joining the thread with a related question. 
What about if after inference I want to pad the output to the original input size (as before CropForegoundd)? Is there a straightforward way of doing this?
Thanks a lot.

**Reply by rijobro** [reply]
The coordinates used for `CropForegroundd` are stored in `data["image_meta_dict"]["foreground_start_coord"]` and `data["image_meta_dict"]["foreground_end_coord"]` by default. You could get these values and then pad by that amount using `BorderPad`.

Alternatively, I'm working on being able to perform the inverse of transforms, which would return the image to the original size. Feel free to try out [the PR](https://github.com/Project-MONAI/MONAI/pull/1663) and use [this tutorial PR](https://github.com/Project-MONAI/tutorials/pull/129) for hints along the way.
